### PR TITLE
silent footsteps for ninja

### PR DESCRIPTION
**Pull request by muburu**
## About the PR
makes ninja footsteps completely silent by making their shoes produce no audible footstep sounds.
doesn't make anything else silent if you're wondering.

## Why / Balance
ninjas having normal footstep sounds that you can easily hear makes them a dead giveaway if you hear them a single time in maints, their abnormally fast footsteps make it super obvious there's a ninja on station. unless someone's been giving away meth, it's 100% a ninja you're hearing.

should hopefully open up more opportunities for cool ninja stealth gameplay, making it much harder to be noticed when you're in a dark area. the not completely invisible cloaking makes them fairly easy to spot them in the open, so this still encourages ninjas to use the darkness to their advantage, effectively disappearing if they run off into the dark.

this was kind of made in response to https://github.com/space-wizards/space-station-14/pull/33244, incentivizing ninjas to actually be stealthy and not just rely on their instant meme stun ability.

## Technical details
just gives ninja shoes a footstep sound collection ~~with volume -100, if this is too shitty of a way to do this or has unforeseen consequences let me know :)~~ with null attached so it draws from no soundpool (silent)
(disclaimer: i am not very smart)

## Media
nothing to see or hear here

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- tweak: Space Ninjas now have silent footsteps.

### CODE REVIEW

**Comment by ArtisticRoomba** [comment]
I mean, makes sense? I think that the fear factor of hearing a ninja prouncing around, but not knowing where it is, is something cool to have. Outright removing their footsteps and enabling them to silently leave and enter, even while uncloaked, feels a bit weird.

IMO I'd like to see the footsteps not making noise as a component of their cloak, but that might be out of scope.

As a stop-gap you could see about making their footsteps be really quiet instead of loud. That way you'd have to be very close to hear them, instead of being able to hear them audibly through a wall running around in maints. 

Also, the -100 volume change is seen as a bit hacky, you might want to find a more elegant solution to this problem.

**Comment by muburu** [comment]
> Also, the -100 volume change is seen as a bit hacky, you might want to find a more elegant solution to this problem.

yeah i agree it's a crappy way to do it but idk how else to do it, i am open to suggestions

**Comment by deltanedas** [comment]
make it nullable and if its null dont play any sound at all

i agree that it should only be silent when cloaked though

**Comment by GroggleG** [comment]
I feel like adding more less-than-lethal options like the ninja net would be a better choice but for now this is nice 👍

**Comment by ArtisticRoomba** [comment]
> I feel like adding more less-than-lethal options like the ninja net would be a better choice but for now this is nice 👍 

Ninja has plenty of less-than-lethal options, but most choose to murderbone and unnecessary kill.

The shock gloves are a viable method to discourage an opponent from fighting and the throwing stars give Ninja a viable method for discouraging/stunning at range.

People just don't know how to be a nuisance other than shock, kill, and leave for dead or RR.

**Comment by baddiePRO** [comment]
>People just don't know how to be a nuisance other than shock, kill, and leave for dead or RR.

we need to teach people the art of trolljitzu (steal the secvend and teleport into space with it) :trollface:

**Comment by deltanedas** [comment]
net that sends you to the dojo was even removed from tg i think

**Comment by Kadeo64** [comment]
good idea in theory

in practice just makes ninja more powerful at murderboning

**Comment by baddiePRO** [comment]
> good idea in theory
> 
> in practice just makes ninja more powerful at murderboning
> 
> 

imo asshole ninjas are gonna murderbone either way, this makes it easier for actually stealthy ninjas to get around.

**Comment by lexaSvarshik** [comment]
Change it to use `null` collection instead of using some sounds, but without volume.
```
 - type: FootstepModifier
    footstepSoundCollection:
      collection: null
```

**Comment by muburu** [comment]
> @muburu Change it to use `null` collection instead of using some sounds, but without volume. Works fine, tested.
> 
> ```
>  - type: FootstepModifier
>     footstepSoundCollection:
>       collection: null
> ```

thanks for the help this is a much cleaner way to do it

**Comment by GroggleG** [comment]
> net that sends you to the dojo was even removed from tg i think

This has left me in shambles for years, but yeah I think an end round honour system might allow people to publicly bully people for murderboning as a ninja.

**Review by TheShuEd — APPROVED** [review]

**Comment by muburu** [comment]
sorry for the late response but i don't know a single thing about c# or how to go about doing the requested change
bit out of my league here

**Comment by TheShuEd** [comment]
I kinda made the sound nullable, but I haven't had a chance to test it in game yet

**Comment by slarticodefast** [comment]
I fixed some stuff.
@TheShuEd Could you check it?

**Review by TheShuEd — APPROVED** [review]
Tested in game, work as intended

**Review by slarticodefast — APPROVED** [review]

**Comment by slarticodefast** [comment]
Allright, good to merge then!

File: Resources/Prototypes/Entities/Clothing/Shoes/specific.yml [hunk 1]
```diff
@@ -136,6 +136,9 @@
     # ninja are masters of sneaking around relatively quickly, won't break cloak
     walkModifier: 1.1
     sprintModifier: 1.3
+  - type: FootstepModifier
+    footstepSoundCollection:
+      collection: null
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Clothing/Shoes/specific.yml:141]: shouldnt the null be on the field above

Comment (slarticodefast) [reply Resources/Prototypes/Entities/Clothing/Shoes/specific.yml:141]: damn
yeah it should be

**Pull request merged**